### PR TITLE
Keyboard rura66

### PR DESCRIPTION
**Pull request by ShinoharaTa**
Please open this pull request again as #13613 has not been fixed.
The fix for #13613 has been merged into this pull request for me.
Please do a code review again.

### CODE REVIEW

File: keyboards/rura66/readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+This is a simple left / right split keyboard with 66 keys. Supports reducing the burden on your shoulders.
+
+* Keyboard Maintainer: [T.Shinohara](https://github.com/ShinoharaTa)
+* Hardware Supported:  The Shiro PCBs, ProMicro supported. Using Cherry MX Profile Switches.
+* Hardware Availability: [Booth (Japanese Site)](https://shino3.booth.pm/items/1444895)
+* 日本国内ではBoothで入手できます: [Booth (Japanese Site)](https://shino3.booth.pm/items/1444895)
+
+![RURA66-USE](https://github.com/ShinoharaTa/keyboards/blob/master/RURA66/manual/images/IMG_0211.jpg?raw=true)
+
+Make for this keyboard default keymap (after setting up your build environment):
```
Comment (drashna) [inline keyboards/rura66/readme.md:16]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/rura66/rev1/config.h [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+#define BACKLIGHT_PIN D3
+#define RGB_DI_PIN D3
+#define RGBLED_NUM 66 // Number of LEDs
+
+#ifdef RGB_MATRIX_ENABLE
+  #define DRIVER_LED_TOTAL RGBLED_NUM
+  #define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+//  #define RGB_MATRIX_KEYRELEASES // reacts to keyreleases (instead of keypresses)
+//  #define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+  #define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
```
Comment (drashna) [inline keyboards/rura66/rev1/config.h:68]:
```suggestion
  #define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/rura66/rev1/config.h [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+//  #define RGB_MATRIX_LED_PROCESS_LIMIT (DRIVER_LED_TOTAL + 4) / 5 // limits the number of LEDs to process in an animation per task run (increases keyboard responsiveness)
+//  #define RGB_MATRIX_LED_FLUSH_LIMIT 16 // limits in milliseconds how frequently an animation will update the LEDs. 16 (16ms) is equivalent to limiting to 60fps (increases keyboard responsiveness)
+  #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 150 // limits maximum brightness of LEDs to 150 out of 255. Higher may cause the controller to crash.
+  #define RGB_MATRIX_HUE_STEP 8
+  #define RGB_MATRIX_SAT_STEP 8
+  #define RGB_MATRIX_VAL_STEP 8
+  #define RGB_MATRIX_SPD_STEP 10
+#endif
```
Comment (drashna) [inline keyboards/rura66/rev1/config.h:77]: No animation modes are enabled here, so no modes except solid color, unless they are explicitly enabled. 

```c
// RGB Matrix Animation modes. Explicitly enabled
// For full list of effects, see:
// https://docs.qmk.fm/#/feature_rgb_matrix?id=rgb-matrix-effects
#    define ENABLE_RGB_MATRIX_ALPHAS_MODS
#    define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#    define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#    define ENABLE_RGB_MATRIX_BREATHING
#    define ENABLE_RGB_MATRIX_BAND_SAT
#    define ENABLE_RGB_MATRIX_BAND_VAL
#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#    define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#    define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#    define ENABLE_RGB_MATRIX_CYCLE_ALL
#    define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#    define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#    define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#    define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#    define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#    define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#    define ENABLE_RGB_MATRIX_DUAL_BEACON
#    define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#    define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#    define ENABLE_RGB_MATRIX_RAINDROPS
#    define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#    define ENABLE_RGB_MATRIX_HUE_BREATHING
#    define ENABLE_RGB_MATRIX_HUE_PENDULUM
#    define ENABLE_RGB_MATRIX_HUE_WAVE
#    define ENABLE_RGB_MATRIX_PIXEL_RAIN
#    define ENABLE_RGB_MATRIX_PIXEL_FLOW
#    define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
// enabled only if RGB_MATRIX_FRAMEBUFFER_EFFECTS is defined
#    define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#    define ENABLE_RGB_MATRIX_DIGITAL_RAIN
// enabled only of RGB_MATRIX_KEYPRESSES or RGB_MATRIX_KEYRELEASES is defined
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#    define ENABLE_RGB_MATRIX_SPLASH
#    define ENABLE_RGB_MATRIX_MULTISPLASH
#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

File: keyboards/rura66/rev1/oled_display.c [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    oled_write_P(qmk_logo, false);
+}
+
+static void render_rgbled_status(bool full) {
+#ifdef RGBLIGHT_ENABLE
+  char buf[30];
+  if (RGBLIGHT_MODES > 1 && rgblight_is_enabled()) {
+      if (full) {
+          snprintf(buf, sizeof(buf), " LED %2d: %d,%d,%d ",
```
Comment (drashna) [inline keyboards/rura66/rev1/oled_display.c:92]: Just a heads up, snprintf can use ~1.5k of firmware space.  

https://docs.qmk.fm/#/squeezing_avr?id=oled-tweaks

File: keyboards/rura66/rules.mk [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+SPLIT_KEYBOARD = yes
+DIP_SWITCH_ENABLE = no
```
Comment (drashna) [inline keyboards/rura66/rules.mk:23]:
```suggestion
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
SPLIT_KEYBOARD = yes
```

File: keyboards/rura66/rura66.h [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#pragma once
```
Comment (drashna) [inline keyboards/rura66/rura66.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by tzarc** [comment]
Please fix your merge such that it's only your changes.

**Comment by ShinoharaTa** [comment]
I'm sorry.
There are many changes.
Delete this branch once, create a new branch from develop, and submit your pull request.
Please close this pull request.

**Comment by drashna** [comment]
rebasing it will most likely fix most of these issues.

**Comment by noroadsleft** [comment]
If rebasing is too intimidating or confusing, you can use `git checkout` to grab the files from another branch/commit, at the cost of losing the original Git history.

```
# Change to your local `Keyboard_RURA66` branch
git checkout Keyboard_RURA66

# Updates your local repo's `master` branch to be identical to that of QMK's `master` branch
git fetch upstream master:master

# Rewrites your local repo's `Keyboard_RURA66` branch to be identical to that of your local `master` branch
git reset --hard master                                                   

# Retrieves the contents of the `keyboards/rura66/` directory as of commit `0cf502fa96d30a643f0607650c628287a39b30ea`
git checkout 0cf502fa96d30a643f0607650c628287a39b30ea -- keyboards/rura66/

# Commit to branch as usual
git commit -m "Add RURA66 new keyboard"

# Push your local `Keyboard_RURA66` branch to GitHub, overwriting the changes to the `Keyboard_RURA66` branch that are on your GitHub fork
git push --force-with-lease origin
```

**Comment by noroadsleft** [comment]
There's a lot of unrelated files in this pull request. With your permission, I can re-implement this PR from a clean state so it's only the files necessary to support this board.

**Comment by ShinoharaTa** [comment]
Thank you for your comments.
I am hoping to be able to resume this pull request from a clean environment.
Please let me know the steps to allow it. I want to complete this keyboard myself.

**Comment by noroadsleft** [comment]
@ShinoharaTa,

The following commands will clean the branch you used for this PR:

```sh
# Untrack the additional files
#         -r = allow recursive removal
#   --cached = removes the files from Git's tracking, but leaves them present on your filesystem
git rm -r --cached keyboards/_mayshino30/ keyboards/blackmask60/ keyboards/mayshi_no30/

# Commit the changes
git commit -m "remove additional files"

# Update your GitHub fork and this PR:
git push
```

You'll be able to re-add the files to Git by creating a new branch or branches, and adding the files to those branches using your normal methods.

**Comment by ShinoharaTa** [comment]
@noroadsleft 

Sorry for the delay in responding.
I ran the command as you pointed out and pushed it to my repository.
I would appreciate you reopening this PR if possible.

File: keyboards/rura66/rev1/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  [_RAISE] = LAYOUT(
+      _______, _______, _______, _______, _______, _______,                                     _______, _______, _______, _______, _______, XXXXXXX,
+      _______, _______, _______, _______, XXXXXXX, XXXXXXX,                                     XXXXXXX, XXXXXXX, _______, XXXXXXX, XXXXXXX, _______,
+      _______, XXXXXXX, _______, _______, _______, XXXXXXX, XXXXXXX,                   XXXXXXX, XXXXXXX, _______, _______, _______, XXXXXXX, _______,
+      _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+      _______, _______, _______,                   _______, _______, XXXXXXX, _______, _______, _______,                   _______, _______, _______
+      ),
+  [_ADJUST] =  LAYOUT(
+      XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,                                     XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+      XXXXXXX, RESET,   RGBRST,  EEP_RST, XXXXXXX, XXXXXXX,                                     RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, XXXXXXX, XXXXXXX,
```
Comment (fauxpark) [inline keyboards/rura66/rev1/keymaps/default/keymap.c:64]:
```suggestion
      XXXXXXX, QK_BOOT, RGBRST,  EE_CLR,  XXXXXXX, XXXXXXX,                                     RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, XXXXXXX, XXXXXXX,
```

File: keyboards/rura66/rev1/info.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+  "keyboard_name": "RURA66",
+  "url": "https://github.com/ShinoharaTa/keyboards/Rura66",
+  "maintainer": "ShinoharaTa",
+  "width": 16,
+  "height": 5,
```
Comment (fauxpark) [inline keyboards/rura66/rev1/info.json:6]:
```suggestion
```

File: keyboards/rura66/rev1/config.h [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline keyboards/rura66/rev1/config.h:160]:
```suggestion
```

File: keyboards/rura66/rev1/config.h [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x3060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    ShinoharaTa
+#define PRODUCT         RURA66
+
```
Comment (fauxpark) [inline keyboards/rura66/rev1/config.h:28]:
```suggestion
```

File: keyboards/rura66/rev1/info.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{
+  "keyboard_name": "RURA66",
```
Comment (fauxpark) [inline keyboards/rura66/rev1/info.json:2]:
```suggestion
  "keyboard_name": "RURA66",
  "manufacturer": "ShinoharaTa",
```

File: keyboards/rura66/rev1/info.json [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  "url": "https://github.com/ShinoharaTa/keyboards/Rura66",
+  "maintainer": "ShinoharaTa",
```
Comment (fauxpark) [inline keyboards/rura66/rev1/info.json:4]:
```suggestion
  "maintainer": "ShinoharaTa",
  "usb": {
    "vid": "0xFEED",
    "pid": "0x3060",
    "device_version": "0.0.1"
  },
```

File: keyboards/rura66/rev1/config.h [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
+ */
+#define SOFT_SERIAL_PIN D2  // or D1, D2, D3, E6
+
+#define BACKLIGHT_PIN D3
+#define RGB_DI_PIN D3
+#define RGBLED_NUM 66 // Number of LEDs
+
+#ifdef RGB_MATRIX_ENABLE
+  #define DRIVER_LED_TOTAL RGBLED_NUM
```
Comment (fauxpark) [inline keyboards/rura66/rev1/config.h:64]:
```suggestion
  #define RGB_MATRIX_LED_COUNT 66
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — COMMENTED** [review]
Your list of included files is now correct. :+1: 

No issues other than the ones already flagged by drashna (https://github.com/qmk/qmk_firmware/pull/15853#pullrequestreview-851067481) and fauxpark (https://github.com/qmk/qmk_firmware/pull/15853#pullrequestreview-1200536269).

**Comment by ShinoharaTa** [comment]
Thanks to all the maintainers for the code check.

I would like to confirm one thing.
Should I write in Data-Driven? Or old style?

I couldn't decide on my own and would like to get your opinion.

**Comment by yswallow** [comment]
I did many changes:

* add instructions to how to enter bootloader on README
* remove `snprintf()` on `oled_display.c`
* disable RGB Matrix and backlight
* disable `SPLIT_HAND_MATRIX_GRID`
* change VID/PID
* add VIA keymap

We're sorry that we enabled unnecessary features and that we make changes after your reviews.

**Review by drashna — APPROVED** [review]

File: keyboards/rura66/rules.mk [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+#   change yes to no to disable
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/rura66/rules.mk:17]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/rura66/rules.mk [hunk 15]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
```
Comment (fauxpark) [inline keyboards/rura66/rules.mk:8]:
```suggestion
#   change yes to no to disable
#
```

File: keyboards/rura66/rules.mk [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/rura66/rules.mk:9]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — CHANGES_REQUESTED** [review]
One small thing from me:

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Comment by ShinoharaTa** [comment]
Thanks for the merge.
I appreciate all of you.

**Pull request merged**